### PR TITLE
WT-2941 format should use faster key-generation functions

### PR DESCRIPTION
**Pull request by keithbostic**
@agorrod, since it made a difference for wtperf, might as well speed up format, too.

### CODE REVIEW

**Comment by keithbostic** [comment]
@ddanderson, there's some code in `cur_json.c:__json_unpack_put()`:
```
        case 'b':
        case 'h':
        case 'i':
        case 'l':
        case 'q':
                return (s +
                    (size_t)snprintf((char *)buf, bufsz, "%" PRId64, pv->u.i));
        case 'B':
        case 't':
        case 'H':
        case 'I':
        case 'L':
        case 'Q':
        case 'r':
        case 'R':
                return (s +
                    (size_t)snprintf((char *)buf, bufsz, "%" PRId64, pv->u.u));
```
that's calling `sprintf` to convert `int64_t` values to strings. We have some functions now that are hugely faster than `sprintf` at doing that work, is there any value in doing faster conversion here?

**Comment by sueloverso** [comment]
I'd say the more consistency the better.  None of these are on performance critical paths anyway but faster is always better.  We would need to move those functions into the src/support directory rather than the test directory.

**Comment by ddanderson** [comment]
@keithbostic, I agree with Sue.  It doesn't seem performance critical, but if we have the routines already, then why not.

File: test/format/util.c [hunk 1]
```diff
@@ -78,19 +78,26 @@ key_gen_setup(WT_ITEM *key)
...
-	 * The key always starts with a 10-digit string (the specified cnt)
-	 * followed by two digits, a random number between 1 and 15 if it's
-	 * an insert, otherwise 00.
+	 * The key always starts with a 10-digit string (the specified row
+	 * number), followed by a suffix if it's an insert.
 	 */
-	len = sprintf(p, "%010" PRIu64 ".%02d", keyno, suffix);
+	u64_to_string_zf(keyno, key->mem, 11);
+	if (suffix == NULL)
+		len = 10;
```
Comment (agorrod) [inline test/format/util.c]: Does it matter that this is functionally different to the previous implementation? The old version always created the same length strings, this one will create strings with a different length depending on whether the suffix is NULL.

Comment (keithbostic) [inline test/format/util.c]: I totally missed that, I thought I'd matched the previous behavior! Thank you!

I pushed a change to get back to the previous behavior. I have no idea if having those two extra bytes is material to how format runs, but I'm not interested in figuring it out, either.

**Comment by agorrod** [comment]
Thanks @keithbostic lgtm. I agree that the json usage isn't on a performance path, so I'm inclined to leave it alone

**Pull request merged**